packages/amplify_api/lib/graphql/graphql_helpers.dart

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api_plugin_interface/amplify_api_plugin_interface.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import '../amplify_api.dart';
+
+class ModelQueriesFactory extends ModelQueriesInterface {
+  // Singleton methods/properties
+  // usage: ModelQueriesFactory.instance;
+  ModelQueriesFactory._();
+
+  static final ModelQueriesFactory _instance = ModelQueriesFactory._();
+
+  static ModelQueriesFactory get instance => _instance;
+
+  @override
+  GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id) {
+    Map<String, String> variableInput = {"id": "ID"};
+
+    return GraphQLRequestFactory().buildQuery<T>(
+        modelType: modelType,
+        variableInput: variableInput,
+        id: id,
+        requestType: GraphQLRequestType.query,
+        requestOperation: GraphQLRequestOperation.get);
+  }
+}
+
+class GraphQLRequestFactory extends GraphQLRequestFactoryInterface {
+  String _getModelType(ModelFieldTypeEnum? val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  String _getFieldsFromModelType(ModelSchema schema) {
+    Map<String, ModelField?>? fieldsMap = schema.fields;
+    return fieldsMap?.entries
+            .map((entry) => entry.value?.association == null ? entry.key : '')
+            .toList()
+            .join(' ') ??
+        '';
```

Comment:
Is the empty string a valid case? If not, should this just throw error if null?

Reply:
Good question. The map's keys are nullable, so code-wise it's valid, but maybe not in practice? I imagine a model will always have fields.

Reply:
I also think a valid model will have these fields, just suggest accounting for if that case is not true in a way caller will know why they are getting this error. If they just see empty string, will be hard to know that it changed and why if they were debugging.

One idea is to add this logic to the validate helper I mentioned, then can just assume this is not null here. Can just manually throw the error here, too.